listenbrainz/webserver/static/js/src/RecentListens.tsx

```diff
@@ -234,7 +234,18 @@ export default class RecentListens extends React.Component<
   };
 
   receiveNewListen = (newListen: string): void => {
-    const listen = JSON.parse(newListen) as Listen;
+    const json = JSON.parse(newListen);
+    // the websocket message received may not contain the expected track_metadata and listened_at fields
+    // therefore, we look for their alias as well.
+    if (!("track_metadata" in json)) {
```

Comment:
I this this could be made a bit more reliable by
1. checking if data is present in the object :`if (!("track_metadata" in json) &&  "data" in json)`. Same goes for `listened_at/timestamp`.
2. add an `else` statement to handle cases where there is neither `track_metadata` nor `data`. We probably want to write a helpful message to `console.debug()` and return
3. Wrap the whole JSON parsing and keys checking in a try-catch block and log any caught error to `console.error()` and return

Just some future proofing to make sure we never get the page breaking for failing at some websocket message processing :)

Reply:
Agreed :). I have updated the PR with the changes.